src/main/scala/mesosphere/mesos/TaskGroupBuilder.scala

```diff
@@ -93,20 +93,27 @@ object TaskGroupBuilder {
     hostPorts: Seq[Option[Int]]): Seq[mesos.NetworkInfo.PortMapping] = {
 
     endpoints.zip(hostPorts).collect {
-      case (mapping, Some(hostPort)) =>
-        if (mapping.containerPort.isEmpty || mapping.containerPort.get == 0) {
-          mesos.NetworkInfo.PortMapping.newBuilder
-            .setHostPort(hostPort)
-            .setContainerPort(hostPort).build
+      case (endpoint, Some(hostPort)) =>
+        val portMapping = mesos.NetworkInfo.PortMapping.newBuilder
+          .setHostPort(hostPort)
+
+        if (endpoint.containerPort.isEmpty || endpoint.containerPort.get == 0) {
+          portMapping.setContainerPort(hostPort)
         } else {
-          val portMapping = mesos.NetworkInfo.PortMapping.newBuilder
-            .setHostPort(hostPort)
-          mapping.containerPort.foreach(portMapping.setContainerPort)
+          endpoint.containerPort.foreach(portMapping.setContainerPort)
+        }
 
-          // TODO(nfnt): set the protocol
-          portMapping.build
+        // While the protocols in RAML may be declared in a list, Mesos expects a
+        // port mapping for every single protocol. If protocols are set, a port mapping
+        // will be created for every protocol in the list.
+        if (endpoint.protocol.isEmpty) {
+          Seq(portMapping.build)
```

Comment:
It is implicitly, by `endpoint.protocol.map`, because `protocol` is a `Seq`. Both code path of the if-else should return `Seq[mesos.NetworkInfo.PortMapping]`.